core/src/main/java/com/datastax/oss/driver/internal/core/metadata/token/DefaultReplicationStrategyFactory.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.core.metadata.token;
+
+import com.datastax.oss.driver.internal.core.context.InternalDriverContext;
+import com.google.common.base.Preconditions;
+import java.util.Map;
+
+public class DefaultReplicationStrategyFactory implements ReplicationStrategyFactory {
+
+  private final String logPrefix;
+
+  public DefaultReplicationStrategyFactory(InternalDriverContext context) {
+    this.logPrefix = context.clusterName();
+  }
+
+  @Override
+  public ReplicationStrategy newInstance(Map<String, String> replicationConfig) {
```

Comment:
It'd be nice to have some implementation of this that doesn't throw an IllegalArgumentException when no match is found, that way one could reuse this logic to evaluate these strategies and than add their additional code for additional strategies they implement.

Reply:
If you test your strategies first and only delegate to `super()` if none matches, it should work no?

Reply:
Oh hah, can't believe I didn't think of that, yeah that does work great :+1: